sdk/metric/correct_test.go

```diff
@@ -588,3 +589,25 @@ func TestSyncInAsync(t *testing.T) {
 		"observer.lastvalue//R=V": 10,
 	}, out.Map())
 }
+
+func TestMeterDoesNotMutateOriginalAttributes(t *testing.T) {
+	ctx := context.Background()
+	meter, sdk, _ := newSDK(t)
+	counter := Must(meter).NewInt64Counter("name.sum")
+
+	labels := []attribute.KeyValue{
+		attribute.String("B", "val"),
+		attribute.String("C", "val"),
+		attribute.String("A", "val"),
+	}
+
+	sdk.RecordBatch(
+		ctx,
+		labels,
+		counter.Measurement(1),
+	)
+
+	assert.Equal(t, "B", string(labels[0].Key))
+	assert.Equal(t, "C", string(labels[1].Key))
+	assert.Equal(t, "A", string(labels[2].Key))
```

Comment:
This assures that `labels` hasn't been modified, do we need any additional assertions that the attributes used by `RecordBatch` have been sorted properly?

Reply:
Added assertions for order of labels in the result after `RecordBatch` has been called.